web/components/application/hacker/pages/PageOne.js

```diff
@@ -1,102 +1,67 @@
-            count={count}
-            activeIndex={activeIndex}
-            lastValidIndex={lastValidIndex}
-            onClick={onPageChange}
-            className="pad-bottom-mega"
-            />
-          <div className="hacker-application-page" id="hacker-application-page-1">
-            <h1>About you <span role="img" aria-label="book">📚</span></h1>
-            <p>
-            nwHacks is Western Canada’s largest collegiate hackathon taking place
-            on January 26th - 27th, 2019 at the University of British Columbia.
-            As expected, we receive a large number of applications every year.
-            We focus on curating a quality hackathon experience for each attendee.
-            For a fair assessment of your application,
-            we encourage you to put your best foot forward on this journey!<span role="img" aria-label="mountain">⛰️</span>
-            </p>
-            <TextInput
-              label="First Name"
-              name="first-name"
-              value={firstName}
-              onChange={(fname) => {
-                const updatedHackerApplication = update(application, {
-                  $merge: {
-                    firstName: fname,
-                  },
-                });
-                onApplicationChange(updatedHackerApplication);
-                this.updateNextButtonState(fname);
-              }}
-              />
-          </div>
-          <ButtonGroup className="pad-top-mega">
-            <SecondaryButton
-              text={activeIndex === 0 ? 'Cancel' : 'Back'}
-              onClick={activeIndex === 0 ? cancelApplication : onPageBack}
-              />
-            <PrimaryButton
-              disabled={!isNextButtonEnabled}
-              text={getPrimaryButtonText(activeIndex, count)}
-              onClick={activeIndex !== count - 1 ? onPageNext : () => console.log('submit application')}
-              />
-          </ButtonGroup>
-        </div>
-      </div>
+            const { isNextButtonEnabled } = this.state;
+
+            return (
+              <HackerApplicationPageTemplate isNextButtonEnabled={isNextButtonEnabled}>
+                <h1>About you <span role="img" aria-label="book">📚</span></h1>
+                <p>
+                nwHacks is Western Canada’s largest collegiate hackathon taking place
+                on January 26th - 27th, 2019 at the University of British Columbia.
+                As expected, we receive a large number of applications every year.
+                We focus on curating a quality hackathon experience for each attendee.
+                For a fair assessment of your application,
+                we encourage you to put your best foot forward on this journey!<span role="img" aria-label="mountain">⛰️</span>
+                </p>
+                <TextInput
+                  label="First Name"
+                  name="first-name"
+                  value={firstName}
+                  onChange={(fname) => {
```

Comment:
Correct me if I'm wrong, but when we add more components to this page, wouldn't a similar `onChange` function be required for every required input field in the page? We should probably make a more generic function that we pass to `onChange` that takes the field name and value as params.

Reply:
good point!!